src/algebraic_topology/dold_kan/faces.lean

```diff
@@ -0,0 +1,218 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import algebraic_topology.dold_kan.homotopies
+import data.nat.parity
+import tactic.ring_exp
+
+/-!
+
+# Study of face maps for the Dold-Kan correspondence
+
+TODO (@joelriou) continue adding the various files references below
+
+In this file, we obtain the technical lemmas that are used in the file
+`projections.lean` in order to get basic properties of the endomorphisms
+`P q : K[X] ⟶ K[X]` with respect to face maps (see `homotopies.lean` for the
+role of these endomorphisms in the overall strategy of proof).
+
+The main lemma in this file is `higher_faces_vanish_induction`. It is based
+on two technical lemmas `Hσφ_eq_neg_σδφ` and `Hσφ_eq_zero`.
+
+-/
+
+open nat
+open category_theory
+open category_theory.limits
+open category_theory.category
+open category_theory.preadditive
+open category_theory.simplicial_object
+open_locale simplicial dold_kan
+
+namespace algebraic_topology
+
+namespace dold_kan
+
+variables {C : Type*} [category C] [preadditive C]
+variables {X : simplicial_object C}
+
+/-- We shall say that a morphism `φ : Y ⟶ X _[n+1]` satisfies `higher_faces_vanish q φ`
+when the compositions `φ ≫ X.δ j` are `0` for `j ≥ max 1 (n+2-q)`. When `q ≤ n+1`,
+it basically means that the composition `φ ≫ X.δ j` are `0` for the `q` highest
+possible values of a nonzero `j`. Otherwise, when `q ≥ n+2`, all the compositions
+`φ ≫ X.δ j` for nonzero `j` vanish. -/
+def higher_faces_vanish {Y : C} {n : ℕ} (q : ℕ) (φ : Y ⟶ X _[n+1]) : Prop :=
```

Comment:
Is this a predicate that also shows up in the usual literature, or is it something you made up to aid formalization?
If it occurs in the literature, please add a pointer.

Reply:
This only appears implicitly in the litterature. I have slightly clarified the meaning of this predicate in the docstring by saying that `higher_faces_vanish q φ` is equivalent to the identity `φ ≫ (P q).f (n+1) = φ` (which in the case of pseudoabelian categories means that `φ` factors through the image of the projection `P q` which shall be constructed in the next PR in this series); for a big enough `q`, this `(P q).f (n+1)` shall be the projection on the normalised subcomplex parallel to the degenerate subcomplex.